migration/src/main/java/com/intershop/customization/migration/dependencies/MarkerCartridgeAnalyzer.java

```diff
@@ -0,0 +1,320 @@
+package com.intershop.customization.migration.dependencies;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.LoggerFactory;
+
+import com.intershop.customization.migration.pfconfigurationfs.MigrateConfigResources;
+
+/**
+ * tHE 3rd step to examine the cartridge dependencies in a migration context.<br/>
+ * 
+ * This class analyzes marker cartridges in top-level cartridges based on a list of breadcrumb lines.
+ * It checks if marker cartridges are used in top-level cartridges where they are not allowed.<br/>
+ * A marker carteidge implements specific funtionality for an application, e.g. 
+ * <code>com.intershop.business:smc</code> for the application <code>imntershop.SMC</code>. 
+ * That's why it  must not appeare in te dependencies of the <code>intershop.EnterpriseBackoffice</code> PP:
+ * 
+ * The analysis is based on predefined properties files that map top-level cartridges to their allowed marker cartridges.
+ */
+public class MarkerCartridgeAnalyzer
+{
+
+    public static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(MigrateConfigResources.class);
+
+    /**
+     * file to keep the cartridge top level assignment results.<br/>
+     */
+    private static Path cartridgeAssignmentResultsFile = 
+        Paths.get(System.getenv("TEMP")+File.separator +"cartridgeAssignmentResults.txt");
```

Comment:
There is no documentation, that the environment variable `$TEMP` has to be declared.